trio/socket.py

```diff
@@ -9,6 +9,62 @@
 from . import _socket
 import sys as _sys
 
+# expose all uppercase names from standardlib socket to trio.socket
+import socket as _stdlib_socket
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = True  # Not precise but only for python 3.5.0
+if TYPE_CHECKING:
```

Comment:
We should add a `# pragma: no cover` in this line. (I guess we should also consider adding it to the list of patterns that automatically trigger no-cover-mode in our `.coveragerc`? That's not urgent though, we can wait until we see how often we use this pattern.)